src/lib/render.ts

```diff
@@ -0,0 +1,85 @@
+import { SvelteComponent, tick } from 'svelte'
+import Container from './Container.svelte'
+import type { ThrelteContext } from '@threlte/core'
+import type { IntersectionEvent } from '@threlte/extras'
+import type { Object3D } from 'three'
+
+const componentCache = new Set<SvelteComponent>()
+
+export const render = (Component: typeof SvelteComponent, { ...options } = {}) => {
+	const component = new Container({
+		target: document.body,
+		props: {
+			component: Component,
+			...options
+		}
+	})
+
+	componentCache.add(component)
+
+	component.$$.on_destroy.push(() => {
+		componentCache.delete(component)
+	})
+
+	const threlteContext = component.$$.context.get('threlte') as ThrelteContext
+
+	// @TODO(mp): Better way to grab interactivity / dispatcher context
+	// const interactivityContext = [...component.$$.context.values()].find((ctx) => ctx.pointer)
+	const dispatcherContext = [...component.$$.context.values()].find((ctx) => ctx.dispatchers)
+
+	return {
+		component,
+		scene: threlteContext.scene,
+		camera: threlteContext.camera,
+		advance: threlteContext.advance,
+		fireEvent: async (
+			object3D: Object3D,
+			event: keyof ThrelteEvents,
+			payload?: IntersectionEvent<typeof event>
+		) => {
+			const eventDispatcher = dispatcherContext.dispatchers.get(object3D)
+			eventDispatcher(event, payload)
+			await tick()
+		},
+		rerender: async (props: Partial<{ component: typeof SvelteComponent }>) => {
+			component.$set(props)
+			await tick()
+		},
+		unmount: () => {
+			if (componentCache.has(component)) {
+				componentCache.delete(component)
+				component.$destroy()
+			}
+		}
+	}
+}
+
+export const cleanup = () => {
+	for (const component of componentCache) {
+		component.$destroy()
+		componentCache.delete(component)
+	}
+}
+
+export const act = async (fn: () => Promise<unknown>) => {
```

Comment:
The act `fn` should be allowed to by sync, too

```suggestion
export const act = async (fn: () => unknown) => {
```

Reply:
Not sure if I understand this one, could you explain a bit more?

Reply:
Oh duh 🤦 was having a moment. I think I understand, this shouldn't throw a type error if someone doesn't wish to handle the promise?